### ISSUE TITLE
[Feature Request] `__rmatmul__` for LazyTensors

### ISSUE THREAD

**Issue by j-wilson** [cornellius-gp/gpytorch#1911]
# 🚀 Feature Request

Implement `__rtmatmul__` for LazyTensors.

## Motivation

LazyTensors must be converted to vanilla Tensors when used by functions that perform matrix multiplies between Tensors (left) and LazyTensors (right). For example, if `A` is a lazy and `K` is not then expressions such as `A @ K @ A.T` will fail.

**Describe the solution you'd like**
I do not know what challenges present themselves here. This may be as simple as adding something like

```
def __rmatmul__(self, other: Tensor) -> Tensor:
    retvalT = self.transpose(-1, -2).matmul(other)
    return retvalT.transpose(-1, -2) if other.ndim > 1 else retvalT
```

**Are you willing to open a pull request?**
Potentially, yes. It would be good to determine the scope of the PR first however.

**Comment by jacobrgardner** [comment]
I think an implementation that basically does the above (although of course also transpose other) would be a perfectly reasonable first pass at this.

The only thing to note is that lazy tensors that represent symmetric positive definite matrices (which is the dominant use case in gpytorch of course) don't even need to have `_transpose_nonbatch` implemented, since obviously `self.transpose_nonbatch(-2, -1)` would just be the same LT as `self`.

That said, I just looked through all of the LTs again, and basically every one I see either implements `_transpose_nonbatch` or is already actually explicitly assuming the matrix is symmetric, e.g., https://github.com/cornellius-gp/gpytorch/blob/4df30488b2e16b46519cb9d34758745c7e2dbb61/gpytorch/lazy/toeplitz_lazy_tensor.py#L52-L53

so the quick implementation may be all we need here.

**Comment by jacobrgardner** [comment]
@j-wilson Yeah, both the implementation and the basic test looks pretty good to me :+1:
